src/tiled/mainwindow.cpp

```diff
@@ -816,6 +817,27 @@ bool MainWindow::saveFileAs()
     return saveFile(fileName);
 }
 
+void MainWindow::saveAll()
+{
+    foreach (MapDocument *mapDoc, mDocumentManager->documents()) {
+        QString fileName(mapDoc->fileName());
+        bool successfullSave = true;
+        QString error;
+        if (fileName.isEmpty()) {
+            successfullSave = saveFileAs();
+            error = tr("Target filename is empty.");
+        } else
+            successfullSave = mapDoc->save(fileName, &error);
```

Comment:
Please do include the `{` and `}` in such cases. Qt style is to leave them out consistently for whole the `if/else` chain, or to include all of them when necessary.